src/AssocList.mo

```diff
@@ -66,69 +115,181 @@ module {
-          switch (find<K, W>(al2, k, keq)) {
+          switch (find<K, W>(map2, k, equal)) {
             case (null) { ?((k, v1), rec(tl)) };
             case (?v2) { rec(tl) };
           };
         };
       };
     };
-    rec(al1);
+    rec(map1);
   };
 
-  /// Transform and combine the entries of two association lists.
+  /// @deprecated
   public func mapAppend<K, V, W, X>(
-    al1 : AssocList<K, V>,
-    al2 : AssocList<K, W>,
-    vbin : (?V, ?W) -> X,
+    map1 : AssocList<K, V>,
+    map2 : AssocList<K, W>,
+    f : (?V, ?W) -> X,
   ) : AssocList<K, X> = label profile_assocList_mapAppend : AssocList<K, X> {
     func rec(al1 : AssocList<K, V>, al2 : AssocList<K, W>) : AssocList<K, X> = label profile_assocList_mapAppend_rec : AssocList<K, X> {
       switch (al1, al2) {
         case (null, null) { null };
-        case (?((k, v), al1_), _) { ?((k, vbin(?v, null)), rec(al1_, al2)) };
-        case (null, ?((k, v), al2_)) { ?((k, vbin(null, ?v)), rec(null, al2_)) };
+        case (?((k, v), al1_), _) { ?((k, f(?v, null)), rec(al1_, al2)) };
+        case (null, ?((k, v), al2_)) { ?((k, f(null, ?v)), rec(null, al2_)) };
       };
     };
-    rec(al1, al2);
+    rec(map1, map2);
   };
 
-  /// Specialized version of `disj`, optimized for disjoint sub-spaces of keyspace (no matching keys).
+  /// Produces a new map by mapping entires in `map1` and `map2` using `f` and
+  /// concatenating the results. Assumes that there are no collisions between
+  /// keys in `map1` and `map2`.
+  ///
+  /// Example:
+  /// ```motoko include=import,initialize
+  /// import { trap } "mo:base/Debug";
+  ///
+  /// // Create map1 = [(0, 10), (1, 11), (2, 12)]
+  /// var map1 : AssocList<Nat, Nat> = null;
+  /// map1 := AssocList.replace(map1, 0, Nat.equal, ?10).0;
+  /// map1 := AssocList.replace(map1, 1, Nat.equal, ?11).0;
+  /// map1 := AssocList.replace(map1, 2, Nat.equal, ?12).0;
+  ///
+  /// // Create map2 = [(4, "14"), (3, "13")]
+  /// var map2 : AssocList<Nat, Text> = null;
+  /// map2 := AssocList.replace(map2, 4, Nat.equal, ?"14").0;
+  /// map2 := AssocList.replace(map2, 3, Nat.equal, ?"13").0;
+  ///
+  /// // Map and append the two AssocLists
+  /// let newMap =
+  ///   AssocList.disjDisjoint<Nat, Nat, Text, Text>(
+  ///     map1,
+  ///     map2,
+  ///     func((v1, v2) : (?Nat, ?Text)) {
+  ///       switch(v1, v2) {
+  ///         case(?v1, null) {
+  ///           debug_show(v1) // convert values from map1 to Text
+  ///         };
+  ///         case(null, ?v2) {
+  ///           v2 // keep values from map2 as Text
+  ///         };
+  ///         case _ {
+  ///           trap "These cases will never happen in mapAppend"
+  ///         }
+  ///       }
+  ///     }
+  ///   );
+  ///
+  /// List.toArray(newMap)
+  /// ```
+  /// Runtime: O(size1 + size2)
+  ///
+  /// Space: O(1)
+  ///
+  /// *Runtime and space assumes that `f` runs in O(1) time and space.
   public func disjDisjoint<K, V, W, X>(
```

Comment:
We need a better name of this, but what? Obviously not in the PR.

Reply:
I think `merge` sounds intuitive. Like you're "merging maps".